corehq/util/django_migrations.py

```
@@ -61,3 +61,14 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
                 app_label, schema_editor, from_state, to_state)
         finally:
             schema_editor.__class__ = DatabaseSchemaEditor
+
+
+class AlterFieldCreateIndexIfNotExists(migrations.AlterField):
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        schema_editor.__class__ = DatabaseSchemaEditorIfNotExists
+        try:
+            super(AlterFieldCreateIndexIfNotExists, self).database_forwards(
+                app_label, schema_editor, from_state, to_state)
+        finally:
+            schema_editor.__class__ = DatabaseSchemaEditor
```

won't this fail silently if it fails to create the index?